bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -596,11 +596,27 @@ void BCStateTran::createCheckpointOfCurrentStateImpl(uint64_t checkpointNumber)
   LOG_INFO(logger_, "Done creating (and persisting) checkpoint of current state!" << KVLOG(checkpointNumber));
 }
 
+const Digest &BCStateTran::computeDefaultRvbDataDigest() const {
+  // RVB Data is empty: initialize defaultRvbDataDigest with input kDefaultInputRvbDataDigest
+  static Digest defaultRvbDataDigest;
+  static std::once_flag calculate_once;
+
+  std::call_once(calculate_once, [&] {
+    static constexpr int kDefaultInputRvbDataDigest = 1;
+    concord::util::digest::DigestUtil::Context ctx;
+    ctx.update(reinterpret_cast<const char *>(&kDefaultInputRvbDataDigest), sizeof(kDefaultInputRvbDataDigest));
+    ctx.writeDigest(defaultRvbDataDigest.getForUpdate());
+    LOG_INFO(logging::getLogger("concord.bft"), KVLOG(defaultRvbDataDigest));
+  });
+  return defaultRvbDataDigest;
+}
+
 void BCStateTran::getDigestOfCheckpointImpl(uint64_t checkpointNumber,
                                             uint16_t sizeOfDigestBuffer,
                                             uint64_t &outBlockId,
                                             char *outStateDigest,
-                                            char *outOtherDigest) {
+                                            char *outResPagesDigest,
```

Comment:
For digest we have a type called Digest. Why are we not using Digest& instead of char* ptr.

Reply:
Well, this is more of an API design discussion.
I didn't write the original API but I can tell you what I think, from my past conversations with, and from how we deal similar code inside ST - the idea is to be able to handle an arbitrary type fo data, although it is not really practical currently, as all calls work on Digest types. 

The same question could have been asked also for the need of sizeOfDigestBuffer.